nixos/modules/hardware/openrgb.nix

```diff
@@ -0,0 +1,59 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.openrgb;
+in
+
+{
+  options.programs.openrgb = {
+    enable = mkEnableOption ''
+      OpenRGB support. Will install udev rules to allow for more devices
+      to be shown, and optionally run OpenRGB as a server.
+    '';
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.openrgb;
+      description = "The port the OpenRGB server will listen.";
+    };
+
+    server.enable = mkEnableOption ''
+      OpenRGB server. OpenRGB will start as a system process, and launching
+      the OpenRGB client will attempt to connect to this server. This is
+      advantageous as users do not need to run OpenRGB with superuser
+      privileges to edit RGB settings in this case.
+    '';
+
+    server.port = mkOption {
+      type = types.port;
+      default = 6742;
+      example = 7575;
+      description = "The port the OpenRGB server will listen.";
+    };
+  };
+
+  config = mkIf (cfg.enable || cfg.server.enable) {
+
+    # enable more RGB devices to be found
+    services.udev.packages = [ cfg.package ];
+
+    environment.systemPackages = [ cfg.package ];
+
+  } // (mkIf cfg.server.enable) {
+    # should be ran as root, to have write access to hardware
+    systemd.services.openrgb = {
+      description = "OpenRGB server";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        ExecStart = "${cfg.package}/bin/openrgb --server --server-port ${toString cfg.server.port}";
+        Restart = "always";
```

Comment:
Does this need to run as root?

Reply:
oh, I could only get this to work as root, but I guess I also have read the documentation since i first added the package ~year ago

Reply:
if there's a way to avoid root, then I'm all for it

Reply:
Really depends on what resources it tries to use, but this crosses over into elaborate hardening options in systemd and maybe some udev options.

Reply:
```
You'll have to enable user access to your SMBus if you don't run as root.

List all SMBus controllers: sudo i2cdetect -l
Note the number for PIIX4, I801, and NCT6775 controllers.
Give user access to those controllers, for instance: sudo chmod 777 /dev/i2c-0
```

Sounds pretty invasive

Reply:
I guess part of the reason for me wanting to create the module, is so that only the daemon needs to be ran as root. But the user client can just be launched without privileges

Reply:
```
DevicePolicy = "closed";
DeviceAllow = [
  "char-i2c rw"
];
```

would be one hardening option that comes to mind.